### PR TITLE
fix: deal damage to unlinked tokens

### LINKED ISSUE #232: Damage doesn't work for unlinked tokens
**Describe the bug**
On an unlinked token, do damage, watch nothing happen.

**To Reproduce**
See above.

**Expected behavior**
Damage should happen on this token, not on any other unlinked copies, or on the base actor for that matter.

**Additional context**
Might have something to do with the code using actor.update, but, checking if linked and if not using token.update didn't help.
Haven't seen any other systems having problems, so not sure what's going on.

### PR DESCRIPTION
**Pull request by jonepatr**
* **Please check if the PR fulfills these requirements**
- [x] We use semantic versioning (https://github.com/semantic-release/semantic-release to be specific), so follow https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#-git-commit-guidelines *EXCEPT*, don't use 'feat:' if you're not me. Stepping major versions until 1.0 is a deliberate decision (after that, it'll be full semantic versioning.) 
- [ ] Docs have been added / updated (for bug fixes / features) - not needed

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
bug fix

* **What is the current behavior?** (You can also link to an open issue here)
Change so that damage can be dealt to both linked  and unlinked tokens.

* **What is the new behavior (if this is a feature change)?**
The change restructures how damage is being dealt and adds a new hook for recalulating the hits when the characteristics change.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

* **Other information**:
Fix #232

### CODE REVIEW

File: src/module/hooks/index.ts [hunk 1]
```diff
@@ -4,4 +4,5 @@ import './ready.ts';
 import './renderChatMessage.ts';
 import './renderItemSheet.ts';
 import './renderSettingsConfig.ts';
+import './updateHits.ts';
```
Comment (jonepatr) [inline src/module/hooks/index.ts:7]: So here I departed from having two files, one for each hook, and instead opted for using more of a "module". But if we want to have one file per hook I can make it so instead.

File: src/module/sheets/TwodsixActorSheet.ts [hunk 2]
```diff
@@ -61,26 +59,8 @@ export class TwodsixActorSheet extends AbstractTwodsixActorSheet {
 
     html.find('.roll-damage').on('click', (this._onRollDamage.bind(this)));
 
-    html.find('.stat-damage').on('change', this._setDamageFromEvent.bind(this));
-    html.find('.special-damage').on('change', this._setDamageFromEvent.bind(this));
   }
 
-  private async _setDamageFromEvent(event:Event):Promise<void> {
```
Comment (jonepatr) [inline src/module/sheets/TwodsixActorSheet.ts:68]: This seems to not be needed? Or am I missing something?

**Comment by xdy** [comment]
Seems to work just fine except for one thing, preexisting damage broke. (E.g. if a character already has 1 point of damage in dex, it get reset to 0 when dragging in more damage.)
I'll fix that (oneliner fix, just change to "let totalDamage = characteristic.damage" inside addDamage so it doesn't get hard set to 0 when the characteristic shouldn't be damaged) and merge to master!

File: src/module/sheets/TwodsixActorSheet.ts (hunk 2 above)
Comment (xdy) [reply src/module/sheets/TwodsixActorSheet.ts:68]: Yeah, this is dead code.

File: src/module/hooks/index.ts (hunk 1 above)
Comment (xdy) [reply src/module/hooks/index.ts:7]: Not sure how I feel about this. One one hand, yeah, a module makes sense. On another, I like being able to easily find the hooks. But, for now at least, sure.

**Comment by xdy** [comment]
Did it via command line, which seems to confuse github. I'll let it run mostly to learn what happens if I do it this way.

**Comment by xdy** [comment]
Yeah, that leaves the PR hanging. Ah well, I'll close it manually.

**Pull request closed without merging**